registry/base_registry.go

```diff
@@ -165,20 +165,29 @@ func (r *BaseRegistry) UnRegister(conf common.URL) error {
 		err    error
 		oldURL common.URL
 	)
-	r.cltLock.Lock()
-	oldURL, ok = r.services[conf.Key()]
-	delete(r.services, conf.Key())
-	r.cltLock.Unlock()
+	func() {
+		r.cltLock.RLock()
+		defer r.cltLock.RUnlock()
+		oldURL, ok = r.services[conf.Key()]
+	}()
+
+	func() {
```

Comment:
look, guy, Number 1, I do not agree with ur modification of the above four lines code block. Number 2, u should not use read lock for a write action(delete). Number 3, u should use RUnlock for RLock.